langchain/llms/openai.py

```diff
@@ -116,9 +116,40 @@ def __call__(self, prompt: str, stop: Optional[List[str]] = None) -> str:
                 response = openai("Tell me a joke.")
         """
         params = self._default_params
+
+        if (params["max_tokens"] == -1): 
+            params["max_tokens"] = self.max_token_for_prompt(prompt)
+
         if stop is not None:
             if "stop" in params:
                 raise ValueError("`stop` found in both the input and default params.")
             params["stop"] = stop
         response = self.client.create(model=self.model_name, prompt=prompt, **params)
         return response["choices"][0]["text"]
+
+    def max_token_for_prompt(self, prompt: str) -> int:
+        """Calculate the maximum number of tokens possible to generate for a prompt.
+
+        Args:
+            prompt: The prompt to pass into the model.
+
+        Returns:
+            The maximum number of tokens to generate for a prompt.
+
+        Example:
+            .. code-block:: python
+
+                max_tokens = openai.max_token_for_prompt("Tell me a joke.")
+        """
+        # import the GPT-3 tokenizer
+        from transformers import GPT2TokenizerFast
```

Comment:
nit: can we do a try/except around this import statement (just to throw a more user friendly error) like in https://github.com/hwchase17/langchain/blob/5267ebce2d9144ef0b530ae04434008fc5618d8d/langchain/text_splitter.py#L64